Formula/hyperkit.rb

```diff
@@ -0,0 +1,45 @@
+class Hyperkit < Formula
+  desc "Toolkit for embedding hypervisor capabilities in your application"
+  homepage "https://github.com/moby/hyperkit"
+  url "https://github.com/moby/hyperkit/archive/v0.20180403.tar.gz"
+  sha256 "e2739b034f20d9437696de48ace42600f55b7213292ec255032b2ef55f508297"
+
+  depends_on "ocaml" => :build
+  depends_on "opam" => :build
+  depends_on :xcode => ["9.0", :build]
+
+  depends_on "libev"
+
+  def install
+    system "opam", "init", "--no-setup"
+    opam_dir = "#{buildpath}/.brew_home/.opam"
+    ENV["CAML_LD_LIBRARY_PATH"] = "#{opam_dir}/system/lib/stublibs:/usr/local/lib/ocaml/stublibs"
+    ENV["OPAMUTF8MSGS"] = "1"
+    ENV["PERL5LIB"] = "#{opam_dir}/system/lib/perl5"
+    ENV["OCAML_TOPLEVEL_PATH"] = "#{opam_dir}/system/lib/toplevel"
+    ENV.prepend_path "PATH", "#{opam_dir}/system/bin"
+
+    inreplace "#{opam_dir}/compilers/4.05.0/4.05.0/4.05.0.comp",
+      '["./configure"', '["./configure" "-no-graph"' # Avoid X11
+
+    ENV.deparallelize { system "opam", "switch", "4.05.0" }
+
+    system "opam", "config", "exec", "--",
+           "opam", "install", "-y", "uri", "qcow", "conduit.1.0.0", "lwt.3.1.0",
+           "qcow-tool", "mirage-block-unix.2.9.0", "conf-libev", "logs", "fmt",
+           "mirage-unix", "prometheus-app"
+
+    args = []
+    args << "GIT_VERSION=#{version}"
+    system "opam", "config", "exec", "--", "make", *args
+
+    bin.install "build/hyperkit"
+    man1.install "hyperkit.1"
+  end
+
+  test do
+    assert_match(version.to_s, shell_output("#{bin}/hyperkit -v 2>&1"))
+    assert_match("kexec: failed to load kernel null",
+                 shell_output("#{bin}/hyperkit -f kexec,null 2>&1", 134))
```

Comment:
This test failed with different messages depending on macos version. It can not boot a VM because test env does not have a CPU that supports EPT (x86 virtualization). 
What can I test instead? Any idea?

Reply:
### high sierra
```
15:31:45 ==> /usr/local/Cellar/hyperkit/0.20180403/bin/hyperkit -v 2>&1
15:31:45 ==> /usr/local/Cellar/hyperkit/0.20180403/bin/hyperkit -f kexec,null 2>&1
15:31:45 sh: line 1: 21605 Abort trap: 6           /usr/local/Cellar/hyperkit/0.20180403/bin/hyperkit -f kexec,null 2>&1
15:31:45 Error: hyperkit: failed
15:31:45 </kexec:\ failed\ to\ load\ kernel\ null/> expected to be =~
15:31:45 <"hv_vm_create failed\n">.
```
### sierra
```
15:31:25 ==> /usr/local/Cellar/hyperkit/0.20180403/bin/hyperkit -f kexec,null 2>&1
15:31:25 Error: hyperkit: failed
15:31:25 <134> expected but was
15:31:25 <1>.
```

Reply:
This seems to test the actual functionality of the virtualisation using hyperkit/xhyve/bhyve, which I do not see as a thing to be tested in the build of a brew package. Unless this test-case is a must (for eg. coverage or something) I would consider this test to be unnecessary.
If you want to test that the hyperkit executable is placed in the correct folder, and is part of the brew PATH, I would rather just do `hyperkit -v` (if that is the same as `xhyve -v`) just to see that the executable will run and return the correct Version values.

Reply:
Look at the `xhyve` formula, that is what they did.
https://github.com/Homebrew/homebrew-core/blob/master/Formula/xhyve.rb

Reply:
As an alternative you could test this instead `hyperkit -f bootrom,null` so it will look for a file and answer "Error opening bootrom "null": No such file or directory". Or you could just test and see that the shell exit code is not 0, instead of asserting that a string in the shell output would always be the same.

Reply:
Also, You do not have to use assert_match to match the text output of shell_output. shell_output by itself is an assertion, which checks for the expected exit value.

Reply:
Thanks for your feedback @artheus, I already made a test to evaluate if version number of hyperkit is correct.
I will give a try to `hyperkit -f bootrom,null`

Reply:
`shell_output` only check the exit code according to documentation
https://www.rubydoc.info/github/Homebrew/brew/master/Homebrew/Assertions#shell_output-instance_method